source/src/cip/ciptypes.h

```diff
@@ -347,11 +347,10 @@ typedef EipStatus (*CipGetSetCallback)(CipInstance *const instance,
  *  @param  message_router_response pointer to response.
  *  @return           status of kEipStatusOk or kEipStatusError on failure
  */
-typedef EipStatus (*CipCallback)(CipInstance *const instance,
-                                 const CipMessageRouterRequest *const
-                                 message_router_request,
-                                 CipMessageRouterResponse *const
-                                 message_router_response);
+typedef EipStatus (*CipCallback)(
+    CipInstance *RESTRICT const instance,
+    CipMessageRouterRequest *message_router_request,
+    CipMessageRouterResponse *message_router_response);
```

Comment:
Please dont change constness. This is an assumption used in most of the infrastructure, and gurarantees that the called function does not move the pointer to a new position.
The position is always moved by the caller based on the decoded bytes information